src/base-apis.js

```diff
@@ -678,6 +678,27 @@ MatrixBaseApis.prototype.setRoomDirectoryVisibilityAppService =
     );
 };
 
+// User Directory Operations
+// =========================
+
+/**
+ * Query the user directory with a term matching user IDs, display names and domains.
+ * @param {object} opts options
+ * @param {string} opts.term the term with which to search.
+ * @param {number=} opts.limit the maximum number of results to return. Defaults to 20.
+ * @return {module:client.Promise} Resolves: an array of results.
+ */
+MatrixBaseApis.prototype.searchUserDirectory = function(opts) {
+    const body = {
+        search_term: opts.term,
+        limit: opts.limit || 20,
```

Comment:
It seems funny to impose a default here. Either a limit is a thing we want in general, in which case the default should be applied on the server - or it's not, in which case we shouldn't impose a default here.

in other words: let's make this reflect the server's behaviour. If the app passes a limit, pass it in the request, else leave it up to the server.